var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -288,6 +290,10 @@ def determine_b2_options(self, spec, options):
         #   https://svn.boost.org/trac/boost/ticket/12496
         if spec.satisfies('%clang'):
             options.extend(['pch=off'])
+            if '+clanglibcpp' in spec:
+                options.extend(['toolset=clang',
+                                'cxxflags="-stdlib=libc++"',
```

Comment:
Would that overwrite user defined `CFLAGS`?

Reply:
The options here are added to the arguments of b2 (or bjam). They won't externally overload an environment variable. I would need to dive deeper to see what boost does with both arguments and environment variables internally.